### PR TITLE
Add 'type' Key to Model Configurations and Some Cleanup

### PR DESCRIPTION
**Pull request by needuv**
# Description

Please add an informative description that covers that changes made by the pull request and link all relevant issues.

If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.

# All SDK Contribution checklist:
- [ ] **The pull request does not introduce [breaking changes]**
- [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [ ] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md).**

## General Guidelines and Best Practices
- [ ] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md##building-and-testing)
- [ ] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: sdk/evaluation/azure-ai-evaluation/azure/ai/evaluation/_evaluators/_coherence/_coherence.py [hunk 1]
```diff
@@ -11,7 +11,7 @@
 
 from azure.ai.evaluation._exceptions import ErrorBlame, ErrorCategory, ErrorTarget, EvaluationException
 
-from ..._common.utils import ensure_api_version_in_aoai_model_config, ensure_user_agent_in_aoai_model_config
+from ..._common.utils import construct_prompty_model_config
 
 try:
```
Comment (ninghu) [inline sdk/evaluation/azure-ai-evaluation/azure/ai/evaluation/_evaluators/_coherence/_coherence.py:16]: I think you missed RetrievalChatEvaluator

**Review by nagkumar91 — APPROVED** [review]

File: sdk/evaluation/azure-ai-evaluation/azure/ai/evaluation/_evaluators/_coherence/_coherence.py (hunk 1 above)
Comment (needuv) [reply sdk/evaluation/azure-ai-evaluation/azure/ai/evaluation/_evaluators/_coherence/_coherence.py:16]: thanks, just added it

File: sdk/evaluation/azure-ai-evaluation/azure/ai/evaluation/_model_configurations.py [hunk 2]
```diff
@@ -6,17 +6,24 @@
 
 
 class AzureOpenAIModelConfigurationBase(TypedDict):
```
Comment (ninghu) [inline sdk/evaluation/azure-ai-evaluation/azure/ai/evaluation/_model_configurations.py:8]: just curious, why need AzureOpenAIModelConfigurationBase and OpenAIModelConfigurationBase?

File: sdk/evaluation/azure-ai-evaluation/azure/ai/evaluation/_common/utils.py [hunk 3]
```diff
@@ -70,18 +71,32 @@ def nltk_tokenize(text: str) -> List[str]:
...
-def ensure_api_version_in_aoai_model_config(
+def parse_model_config_type(
     model_config: Union[AzureOpenAIModelConfiguration, OpenAIModelConfiguration],
-    default_api_version: str,
 ) -> None:
     if "azure_endpoint" in model_config or "azure_deployment" in model_config:
-        model_config["api_version"] = model_config.get("api_version", default_api_version)
+        model_config["type"] = AZURE_OPENAI_TYPE
+    else:
+        model_config["type"] = OPENAI_TYPE
```
Comment (ninghu) [inline sdk/evaluation/azure-ai-evaluation/azure/ai/evaluation/_common/utils.py:80]: I thought the type will be set when initializing the model config. Why do we need this logic here?

**Review by ninghu — APPROVED** [review]

**Pull request merged**